appeals/web/src/server/appeals/appeal-details/service-user/service-user.controller.js

```diff
@@ -0,0 +1,92 @@
+import { getAppealDetailsFromId } from '#appeals/appeal-details/appeal-details.service.js';
+import logger from '#lib/logger.js';
+import { addNotificationBannerToSession } from '#lib/session-utilities.js';
+import { changeServiceUserPage } from './service-user.mapper.js';
+import { updateServiceUser } from './service-user.service.js';
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getChangeServiceUser = async (request, response) => {
+	return renderChangeServiceUser(request, response);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderChangeServiceUser = async (request, response) => {
+	const {
+		errors,
+		params: { appealId, userType }
+	} = request;
+
+	const appealDetails = await getAppealDetailsFromId(request.apiClient, appealId);
+	if (userType in appealDetails) {
+		const mappedPageContents = changeServiceUserPage(
+			appealDetails,
+			request.session.updatedServiceUser,
+			userType,
+			errors
+		);
+
+		return response.render('patterns/change-page.pattern.njk', {
+			pageContent: mappedPageContents,
+			errors
+		});
+	}
+
+	return response.render('app/500.njk');
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const postChangeServiceUser = async (request, response) => {
+	request.session.updatedServiceUser = {
+		firstName: request.body['firstName'],
```

Comment:
Just a thought, is it worth checking these `body` values exist before saving them in the session object?

Reply:
They are checked as part of validation as they aren't all needed.